packages/composites/dxdao/src/endpoint/tvl.ts

```diff
@@ -0,0 +1,55 @@
+import { Validator, Logger } from '@chainlink/ea-bootstrap'
+import { AdapterRequest } from '@chainlink/types'
+import { ethers, BigNumber } from 'ethers'
+import * as TokenAllocation from '@chainlink/token-allocation-adapter'
+import { Config } from '../config'
+
+export const NAME = 'TVL'
+
+const customParams = {
+  pairContractAddress: true,
+}
+
+const dxdWethContractAbi = [
+   {
+      "type":"function",
+      "stateMutability":"view",
+      "payable":false,
+      "outputs":[
+         {
+            "type":"uint256",
+            "name":""
+         }
+      ],
+      "name":"balanceOf",
+      "inputs":[
+         {
+            "type":"address",
+            "name":"_owner"
+         }
+      ],
+      "constant":true
+   }
+]
+
+export const getTokenAllocations = async (request: AdapterRequest, config: Config): Promise<TokenAllocation.types.TokenAllocation[]> => {
+   const validator = new Validator(request, customParams)
+   if (validator.error) throw validator.error
+   const wethContractAddress = config.wethContractAddress
+   const { pairContractAddress } = validator.validated.data
+   const tvlInWei = await getTvlAtAddressInWei(pairContractAddress, wethContractAddress, config.rpcUrl)
+   return [{
+      symbol: "WETH",
+      balance: tvlInWei.toString(),
+      decimals: 18
+   }]
+}
+
+const getTvlAtAddressInWei = async (pairContractAddress: string, wethContractAddress: string, jsonRpcUrl: string): Promise<BigNumber> => {
+   const provider = new ethers.providers.JsonRpcProvider(jsonRpcUrl)
+   Logger.info(`Fetching TVL for contract '${pairContractAddress}' using WETH contract address ${wethContractAddress}`)
+   const contract = new ethers.Contract(wethContractAddress, dxdWethContractAbi, provider)
+   const { _hex: pairBalanceHex } = await contract.balanceOf(pairContractAddress)
+   const tvlInWei = BigNumber.from(pairBalanceHex).mul(2)
```

Comment:
Does `.balanceOf()` not already return a BigNumber?

Would this work?

```suggestion
   const tvlInWei = (await contract.balanceOf(pairContractAddress)).mul(2)
```

Reply:
Apparently not.  I had tried this and got an error saying that `.mul` is not a function.

Reply:
Can you do `... as BigNumber`? I'm pretty sure the `_hex` you're getting is the hex from the BigNumber.

At least the ethers.js documentation says so: https://docs.ethers.io/v5/single-page/#/v5/api/contract/contract/-%23-Contract-functionsCall